libethereum/BlockChainSync.cpp

```diff
@@ -522,6 +559,16 @@ void BlockChainSync::onPeerBlockHeaders(std::shared_ptr<EthereumPeer> _peer, RLP
 	continueSync();
 }
 
+bool BlockChainSync::verifyDaoChallengeResponse(RLP const& _r)
+{
+	if (_r.itemCount() != 1)
+		return false;
+
+	BlockHeader info(_r[0].data(), HeaderData);
+	return info.number() == host().chain().sealEngine()->chainParams().daoHardforkBlock &&
+		info.extraData() == fromHex("0x64616f2d686172642d666f726b");
```

Comment:
I think the block hash is harder to imitate.

Reply:
We don't know block hash here, having only header data...

extraData checking of DAO fork block is done also by geth, and also already by our code in `Ethash::verify()`

Reply:
Oh, I see.